central/vulnmgmt/vulnerabilityrequest/datastore/datastore_impl.go

```diff
@@ -125,6 +129,23 @@ func (ds *datastoreImpl) UpdateRequestStatus(ctx context.Context, id, message st
 		return nil, errors.Errorf("vulnerability request %s not found", id)
 	}
 
+	if status == storage.RequestStatus_APPROVED && utils.V2GlobalScope(req.GetScope()) {
+		// Approvers of exceptions affecting all images should have access to all namespaces
+		allResults, err := ds.namespaceDatastore.Search(allAccessCtx, search.EmptyQuery())
+		if err != nil {
+			return nil, err
+		}
+		allIDs := search.ResultsToIDSet(allResults)
+		scopedResults, err := ds.namespaceDatastore.Search(ctx, search.EmptyQuery())
+		if err != nil {
+			return nil, err
+		}
+		scopedIDs := search.ResultsToIDSet(scopedResults)
+		if !allIDs.Equal(scopedIDs) {
+			return nil, errors.Wrapf(errox.NotAuthorized, "Access scope of global exception request approvers should include all namespaces")
```

Comment:
Could you just do an ID query passing in `allIDs` and make sure the result size matched `len(allIDs)`?

Reply:
Good idea.

Reply:
I can also do a count with both contexts and just compare the count.

Reply:
Since we are passing the ID queries, yes probably so